comp/remote-config/rcclient/rcclient.go

```diff
@@ -114,9 +130,57 @@ func (rc rcClient) Start(agentName string) error {
 
 	rc.client.Start()
 
+	if rc.clientHA != nil {
+		rc.clientHA.SetAgentName(agentName)
+		rc.clientHA.Subscribe(state.ProductAgentFailover, rc.haUpdateCallback)
+		rc.clientHA.Start()
+	}
+
 	return nil
 }
 
+func (rc rcClient) haUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	for cfgPath, update := range updates {
+		haUpdate, err := parseHighAvailabilityConfig(update.Config)
+		if err != nil {
+			pkglog.Errorf("HA update unmarshal failed: %s", err)
+			applyStateCallback(cfgPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: err.Error(),
+			})
+			continue
+		}
+
+		// Accept and apply only the first valid update. Disregard any remaining, which will be left in the UNKNOWN state
+		if haUpdate != nil && haUpdate.Failover != nil {
+			pkglog.Infof("Setting `ha.failover: %t` through remote config", *haUpdate.Failover)
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 Praise:** ‏That's a good thing to log as INFO, it'll tell customers that things are working